internal/output/html.go

```diff
@@ -335,11 +381,89 @@ func buildHTMLResult(ecosystemMap map[string][]HTMLSourceResult, resultCount HTM
 	})
 
 	ecosystemResults = append(ecosystemResults, osResults...)
+	var layers []LayerInfo
+	if isContainerScanning {
+		layers = getAllLayers(ecosystemResults)
+	}
+	vulnTypeCount := getVulnTypeCount(ecosystemResults)
 
 	return HTMLResult{
-		EcosystemResults: ecosystemResults,
-		HTMLVulnCount:    resultCount,
+		EcosystemResults:    ecosystemResults,
+		HTMLVulnCount:       resultCount,
+		IsContainerScanning: isContainerScanning,
+		AllLayers:           layers,
+		VulnTypeCount:       vulnTypeCount,
+	}
+}
+
+func getVulnTypeCount(result []HTMLEcosystemResult) VulnTypeCount {
+	osType := 0
+	projectType := 0
+	uncalled := 0
+	all := 0
+
+	for _, ecosystem := range result {
+		for _, source := range ecosystem.Sources {
+			if ecosystem.IsOS {
+				osType += source.HTMLVulnCount.Called
+			} else {
+				projectType += source.HTMLVulnCount.Called
+			}
+			all += source.HTMLVulnCount.Called
+			uncalled += source.HTMLVulnCount.Uncalled
+		}
+	}
+
+	return VulnTypeCount{
+		All:      all,
+		OS:       osType,
+		Project:  projectType,
+		Uncalled: uncalled,
+	}
+}
+
+func getAllLayers(result []HTMLEcosystemResult) []LayerInfo {
+	layerMap := make(map[string]string)
+	layerCount := make(map[string]HTMLVulnCount)
+	layerIndex := 0
+
+	for _, ecosystem := range result {
+		for _, source := range ecosystem.Sources {
+			for _, packageInfo := range source.PackageResults {
+				layerID := packageInfo.HTMLPackageDetail.LayerID
+				layerCommand := packageInfo.HTMLPackageDetail.LayerCommand
+
+				// Check if this layer ID and command combination is already in the map
+				if _, ok := layerMap[layerID]; !ok {
+					var resultCount HTMLVulnCount
+					updateCount(&resultCount, &packageInfo.HTMLVulnCount)
+					layerMap[layerID] = layerCommand // Store the layer ID and command
+					layerCount[layerID] = resultCount
+					layerIndex++
+				} else {
+					resultCount := layerCount[layerID]
+					updateCount(&resultCount, &packageInfo.HTMLVulnCount)
+					layerCount[layerID] = resultCount
+				}
+			}
+		}
 	}
+
+	// Convert the map to a slice of LayerInfo
+	layers := make([]LayerInfo, 0, len(layerMap))
+	i := 0
+	for layerID, layerCommand := range layerMap {
```

Comment:
issue: I feel like this could be a bug, as Go purposely uses a random order when iterating over maps

I would expect currently repeatedly scanning the same image will give you a random layer order each time, unless I've missed a sort somewhere?

(that also means your index is kind of useless)